src/jmclient/wallet.py

```diff
@@ -1062,8 +1063,8 @@ def is_known_script(self, script):
         return script in self._script_map
 
     def get_addr_mixdepth(self, addr):
-        script = self.addr_to_script(addr)
-        return self.get_script_mixdepth(script)
+        path = self.addr_to_path(addr)
```

Comment:
Checking my understanding: by switching to a path lookup, instead of a script lookup, you get to take benefit of the cache where possible, and perhaps also, avoid the expense of the conversion operation in the bitcointx lib?

Reply:
`addr_to_script` is now implemented as a lookup of the path for the address in `_addr_map`, followed by a lookup of the script for the path in `_cache`. Then `get_script_mixdepth` is implemented as a lookup of the path for the script in `_script_map`, followed by extraction of the mixdepth from the path using `_get_mixdepth_from_path`. That's a needlessly wasteful sequence of operations, as you're doing two dictionary lookups plus a cache lookup when you already got the pertinent piece of information (the path) from the first lookup.

Reply:
The cache lookup would always hit since populating `_script_map` and `_addr_map` required calls, respectively, to `get_script_from_path` and `get_address_from_path`, which cache their results. This change is only because one dictionary lookup is faster than two dictionary lookups and one cache lookup (which in turn requires several string encodings and dictionary lookups, the number of which depending on the path length).